### PR TITLE
BAEL-4243:Once per request filter

### PR DESCRIPTION
**Pull request by nishitgoyal17**

### CODE REVIEW

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/OncePerRequestFilterApp.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.baeldung.OncePerRequestFilter;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+
+@SpringBootApplication(scanBasePackages = "com.baeldung.OncePerRequestFilter")
```
Comment (KevinGilmore) [inline spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/OncePerRequestFilterApp.java:7]: correct this while changing the package name

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class MyOncePerRequestFilter extends OncePerRequestFilter {
+    private Logger logger = LoggerFactory.getLogger(MyOncePerRequestFilter.class);
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        logger.info("Inside OncePer Request Filter originated by request {}", request.getRequestURI());
+        filterChain.doFilter(request, response);
+    }
+
+    protected boolean shouldNotFilterAsyncDispatch() {
```
Comment (KevinGilmore) [inline spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java:25]: Use the Override annotation here, too.

**Review by KevinGilmore — COMMENTED** [review]
Package name, Override missing, blank lines, indentation.

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        return true;
+    }
+
+
```
[the remark this answers is not in the document]
Comment (nishitgoyal17) [reply spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java]: done

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/OncePerRequestFilterApp.java (hunk 1 above)
Comment (nishitgoyal17) [reply spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/OncePerRequestFilterApp.java]: done

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
```
[the remark this answers is not in the document]
Comment (nishitgoyal17) [reply spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/OncePerRequestFilter/MyOncePerRequestFilter.java]: done

**Pull request merged**